managed-ledger/src/main/java/org/apache/bookkeeper/mledger/ManagedCursor.java

```diff
@@ -131,6 +131,22 @@ Entry getNthEntry(int n, IndividualDeletedEntries deletedEntries)
     void asyncGetNthEntry(int n, IndividualDeletedEntries deletedEntries, ReadEntryCallback callback,
             Object ctx);
 
+
+    /**
+     * Asynchronously get a message by its id.
+     *
+     * @param ledgerId
+     *              ledger id
+     * @param entryId
+     *              entry id
+     * @param deletedEntries
+     *              skip individual deleted entries
+     * @param callback
+     * @param ctx
+     */
+    void asyncGetMessageById(long ledgerId, long entryId, IndividualDeletedEntries deletedEntries,
```

Comment:
According to what we discussed before, We don't need to specify a subscription name for getting a message by ID. So I think we can get a message from ManagedLedger directly by https://github.com/apache/pulsar/blob/master/managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerImpl.java#L1494. So that we don't need to create a one-time subscription.